File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> mp = new HashMap<Integer,Integer>();
        int[] ans = new int[2];
        for(int i = 0; i < nums.length; i += 1) {
            if(mp.containsKey(target - nums[i])) {
                ans[0] = mp.get(target - nums[i]);
                ans[1] = i;
            }
            mp.put(nums[i],i);
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    pair<int,int> getTwoIndicesThatSumsUpToTarget(vector<int> nums, int target) {
        unordered_map<int,int>  mp;
        int index = 0;
        for(int num : nums) {
            if(mp.count(target - num)) {
                return {mp[target - num], index};
            }
            mp[num] = index;
            index += 1;
        }
        return {-1,-1};
    }

    vector<int> twoSum(vector<int>& nums, int target) {
        pair<int,int> ans = getTwoIndicesThatSumsUpToTarget(nums,target);
        return {ans.first,ans.second};
    }
};